frontend/src/pages/action.js

```diff
@@ -0,0 +1,115 @@
+import React,{useEffect} from "react";
+import { makeStyles, withStyles, Typography, Box, Button, Grid } from "@material-ui/core";
+import Checkbox from "./components/checkbox";
+import { BASE_URL, STUDENT_ID, SCENARIO_ID } from "../constants/config";
+import axios from 'axios';
+import { ScenariosContext } from "../Nav";
+
+// const useStyles = makeStyles((theme) => ({
+//   root: {
+//     backgroundColor: theme.palette.background.paper,
+//     color: "#5b7f95"
+//   },
+// }));
+
+const TextTypography = withStyles({
+  root: {
+    color: "#373a3c"
+  }
+})(Typography);
+
+function Action({ pages, setPages, activePage, setActivePage, content_url, nextPageID, prevPageID, title }) {
+  function goToPage(pageID) {
+    if (pages[pageID].completed) {
+      if (!pages[pageID].visited) {
+        setPages((prevPages) => {
+          let copy = { ...prevPages };
+          copy[pageID].visited = true;
+          return copy;
+        });
+      }
+      setActivePage((prevPage) => pageID);
+    }
+  }
+
+  const [actionQuestion, setActionQuestion, setActionChoices] = React.useState('');
+  const [questionID, setQuestionID] = React.useState('');
+  const [scenarios, setScenarios] = React.useContext(ScenariosContext);
+
+  useEffect(() => {
+    // backend call
+    (async () => {
+      axios({
+        method: 'get',
+        url: BASE_URL + content_url,
+        headers: {
+          scenarioID: scenarios.currentScenarioID,
+          studentID: STUDENT_ID,
+        }
+      }).then(response => {
+        console.log(response);
+        if (scenarios.currentScenarioID == 1)
+        {
+          setActionQuestion(text => response.data[0].question);
+          setQuestionID(id => response.data[0].option_id);
+        }
+        if (scenarios.currentScenarioID == 2){
+          setActionQuestion(text => response.data[1].question);
+          setQuestionID(id => response.data[1].option_id);
+        }
+      }).catch((err)=>{
+        console.log("err",err);
+        //alert(err);
+      });
+    })();
+  }, [scenarios]);
+
+  async function handleResponse(data) {
+    const request_data = {}
+    console.log("Question ID's " + questionID);
+    request_data[questionID[0].toString()] = data;
+    await axios({
+      url: BASE_URL + content_url,
+      method: 'put',
+      data: {
+        scenarioID: scenarios.currentScenarioID,
+        studentID: STUDENT_ID,
+        data: request_data
+      }
+    });
+  }
```

Comment:
When I select the "Approve" option for the final action, It sends `data: {3: "3"}` even though question 3 doesn't have an option with id: 3

Reply:
My bad, this is incorrect. I was looking at the initial action GET.